agent/consul/federation_state_replication.go

```diff
@@ -4,11 +4,28 @@ import (
 	"context"
 	"fmt"
 	"sort"
+	"strings"
 	"time"
 
 	"github.com/hashicorp/consul/agent/structs"
 )
 
+const federationStatsNotSupportedMessage = "Not all servers in the datacenter support federation states - preventing replication"
+
+var (
+	federationStatesNotSupported = federationStatesNotSupportedError{}
+)
+
+type federationStatesNotSupportedError struct{}
+
+func (e federationStatesNotSupportedError) Error() string {
+	return federationStatsNotSupportedMessage
+}
+
+func areFederationStatesNotSupportedError(err error) bool {
+	return strings.Contains(err.Error(), federationStatsNotSupportedMessage)
```

Comment:
`isErrFederationStatesNotSupported` ? The states are plural, but the thing thing being compared is a single error